src/wp-includes/class-wp-term-query.php

```diff
@@ -636,32 +637,20 @@ public function get_terms() {
 
 		$selects = array();
 		switch ( $args['fields'] ) {
-			case 'all':
 			case 'all_with_object_id':
-			case 'tt_ids':
-			case 'slugs':
-				$selects = array( 't.*', 'tt.*' );
-				if ( 'all_with_object_id' === $args['fields'] && ! empty( $args['object_ids'] ) ) {
+				$selects = array( 't.term_id' );
+				if ( ! empty( $args['object_ids'] ) ) {
 					$selects[] = 'tr.object_id';
 				}
 				break;
-			case 'ids':
-			case 'id=>parent':
-				$selects = array( 't.term_id', 'tt.parent', 'tt.count', 'tt.taxonomy' );
-				break;
-			case 'names':
-				$selects = array( 't.term_id', 'tt.parent', 'tt.count', 't.name', 'tt.taxonomy' );
-				break;
 			case 'count':
 				$orderby = '';
 				$order   = '';
 				$selects = array( 'COUNT(*)' );
 				break;
-			case 'id=>name':
-				$selects = array( 't.term_id', 't.name', 'tt.parent', 'tt.count', 'tt.taxonomy' );
-				break;
-			case 'id=>slug':
-				$selects = array( 't.term_id', 't.slug', 'tt.parent', 'tt.count', 'tt.taxonomy' );
+			default:
+				$selects = array( 't.term_id' );
+				$groupby = 'tt.term_taxonomy_id';
```

Comment:
Can you say more about the requirement for `GROUP BY`? I tried running the tests without it, and I see that there are a few failures, but I don't fully understand why - prior to this change, what is the mechanism that's ensuring uniqueness? Here are the before and after queries:

Before: `SELECT  t.*, tt.* FROM wptests_terms AS t  INNER JOIN wptests_term_taxonomy AS tt ON t.term_id = tt.term_id INNER JOIN wptests_term_relationships AS tr ON tr.term_taxonomy_id = tt.term_taxonomy_id WHERE tt.taxonomy IN ('category') AND tr.object_id IN (5)`

After: `SELECT  t.term_id FROM wptests_terms AS t  INNER JOIN wptests_term_taxonomy AS tt ON t.term_id = tt.term_id INNER JOIN wptests_term_relationships AS tr ON tr.term_taxonomy_id = tt.term_taxonomy_id WHERE tt.taxonomy IN ('category') AND tr.object_id IN (5) GROUP BY tt.term_taxonomy_id`

Does `SELECT t.*,tt*` combine duplicates somehow?

I am somewhat concerned that introducing `GROUP BY` to most calls to `get_terms()` will introduce unforeseen performance issues. But maybe this is not a concern due to the `term_taxonomy_id` column. I leave it to the experts :-D

A small stylistic question: there's already a `$distinct` mechanism in `WP_Term_Query` https://core.trac.wordpress.org/browser/tags/5.9.1/src/wp-includes/class-wp-term-query.php?marks=633#L630, which has the same effect. I wonder if there's a reason to standardize on one rather than the other.

Reply:
I honestly don't know. I don't understand how the terms were unique in the first place. I think that this how something to do with it. 

> Does SELECT t.*,tt* combine duplicates somehow?

In my experience, group by improves performances of queries. When looking at other query classes, they all had group bys in them 

https://github.com/WordPress/wordpress-develop/blob/73fdc6ada28aa08815e3614c3373ae590589d05b/src/wp-includes/class-wp-comment-query.php#L936

https://github.com/WordPress/wordpress-develop/blob/73fdc6ada28aa08815e3614c3373ae590589d05b/src/wp-includes/class-wp-query.php#L2849

https://github.com/WordPress/wordpress-develop/blob/73fdc6ada28aa08815e3614c3373ae590589d05b/src/wp-includes/class-wp-site-query.php#L661

If anything adding this group by adds in some constancy here. 

I tried using `$distinct`, but I had the same unit test fail. Group by feels safer here, but I maybe missing something.

Reply:
Added in distinct only where needed and removed group by logic.